src/main/flight/imu.c

```diff
@@ -541,18 +593,31 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
         useMag = true;
     }
 #endif
+
 #if defined(USE_GPS)
-    if (!useMag && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT && gpsSol.groundSpeed >= GPS_COG_MIN_GROUNDSPEED) {
-        // Use GPS course over ground to correct attitude.values.yaw
-        const float imuYawGroundspeed = fminf (gpsSol.groundSpeed / GPS_COG_MAX_GROUNDSPEED, 2.0f);
-        courseOverGround = DECIDEGREES_TO_RADIANS(gpsSol.groundCourse);
-        cogYawGain = (FLIGHT_MODE(GPS_RESCUE_MODE)) ? gpsRescueGetImuYawCogGain() : imuYawGroundspeed;
-        // normally update yaw heading with GPS data, but when in a Rescue, modify the IMU yaw gain dynamically
-        if (shouldInitializeGPSHeading()) {
-            // Reset our reference and reinitialize quaternion.
-            // shouldInitializeGPSHeading() returns true only once.
+    if (!useMag && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT) {
+        static bool gpsHeadingInitialized = false;
+        if (gpsHeadingInitialized) {
+            courseOverGround = DECIDEGREES_TO_RADIANS(gpsSol.groundCourse);
+            // used in imuMahonyAHRSupdate()
+            if (FLIGHT_MODE(GPS_RESCUE_MODE)) {
+                // GPS_Rescue adjusts groundspeedGain during a rescue in a range 0 - 4.5, depending on GPS Rescue state and groundspeed relative to speed to home.
+                groundspeedGain = gpsRescueGetImuYawCogGain();
+            } else {
+                // in normal flight, IMU should:
+                // - heavily average GPS heading values at low speed, since they are random, almost
+                // - respond more quickly at higher speeds.
+                // GPS typically returns quite good heading estimates at or above 0.5- 1.0 m/s, quite solid by 2m/s
+                // groundspeedGain will be 0 at 0.0m/s, rising slowly towards 1.0 at 1.0 m/s, and reaching max of 10.0 at 10m/s
+                const float speedRatio = (float)gpsSol.groundSpeed / GPS_COG_MIN_GROUNDSPEED;
+                groundspeedGain = speedRatio > 1.0f ? fminf(speedRatio, 10.0f) : speedRatio * speedRatio;
```

Comment:
Yes, and no...
We get quite some useful information from about 0.5m/s with modern GPS units.
It gets noisy and random below about that point, and yes, wind drift is going to be a significant issue at low speeds.  
However:
- When flat on pitch, there will be zero ez_ef, so no gain, and no noise issue
- With any significant roll angle, we might get sideways speed, but ez_ef will also be multiplied to zero.
- The IMU code itself smooths the noise at low speed very effectively
- If we start collecting data at low speeds, the heading can correct itself sooner after takeoff
- if we landed, even if pitched forward leading to non-zero ez_ef, the velocity factor would be zero, so ez_ef would be zero.
- in practice we need to be moving quite a lot faster than 1m/s before the PR code would allow the GPS to significantly modify heading.
So I think that, speaking practically, it is OK as is.  
I will do some specific tests to confirm that it's OK, but so far there's no problem.

Reply:
At present the only way to trip up in my testing is to go slightly non-flat and fly slowly at a non-forward angle.  Eg roll just a little bit, and pitch forward barely at all.  Or, in a strong side-wind, if pitched forward only a little, so that the wind strongly determines course over ground direction, then we will get an incorrect heading.  I don't think we can fix these issues, and they already exist.
Overall my testing is complete.